src/test_l1_dsHdmiIn.c

```diff
@@ -1649,21 +1694,22 @@ void test_l1_dsHdmiIn_negative_dsIsHdmiARCPort_sink(void) {
  * 
  */
 void test_l1_dsHdmiIn_positive_dsGetEDIDBytesInfo_sink(void) {
+
     gTestID = 33;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     uint8_t numInputPorts = 0;
     unsigned char* edidBytes1 = (unsigned char*) malloc(255);
     unsigned char* edidBytes2 = (unsigned char*) malloc(255);
     unsigned char* edidBytes3 = (unsigned char*) malloc(255);
     unsigned char* edidBytes4 = (unsigned char*) malloc(255);
```

Comment:
1.Use array's , local structure element instead of malloc everywhere in the file
2. Instead of checking the each port. Loop the ports and validate and at the end validate the same port twice and compare. Attached sample for reference.

"	int allmSupport[3] = {0};

	// Step 3: Loop through all valid HDMI input ports and check ALLM support
        for (uint8_t port = 0; port < numInputPorts; ++port) {
            result = dsGetEdid2AllmSupport(port, &allmSupport[port]);
            UT_ASSERT_EQUAL(result, dsERR_NONE);
        }

        // Validate the last port twice and compare the results
        if (numInputPorts > 1) {
            int lastPortSupport1 = 0;
            int lastPortSupport2 = 0;

            result = dsGetEdid2AllmSupport(numInputPorts - 1, &lastPortSupport1);
            UT_ASSERT_EQUAL(result, dsERR_NONE);

            result = dsGetEdid2AllmSupport(numInputPorts - 1, &lastPortSupport2);
            UT_ASSERT_EQUAL(result, dsERR_NONE);

            // Compare the results for the last port
            UT_ASSERT_EQUAL(lastPortSupport1, lastPortSupport2);
        }"

3. wherever the function description in the interface file says it only supports for the Source always. Use the gsourcetype to compare the source/sink supports

Reply:
Done